flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/operations/SqlNodeConvertContext.java

```diff
@@ -60,7 +60,9 @@ public RelRoot toRelRoot(SqlNode sqlNode) {
     public String toQuotedSqlString(SqlNode sqlNode) {
         SqlParser.Config parserConfig = flinkPlanner.config().getParserConfig();
         SqlDialect dialect =
-                new CalciteSqlDialect(
+                // The default implementation of SqlDialect suppresses all table hints since
+                // CALCITE-4640, so we should use AnsiSqlDialect instead to reserve table hints.
+                new AnsiSqlDialect(
```

Comment:
The difference between `CalciteSqlDialect` and `AntiSqlDialect` is the method `withIdentifierQuoteString()`. Would replacing `\"`  to ```  have any other impact in Flink?

Reply:
The impact of this change is limited to the sql to operation conversion, and will not affect the planner’s own dialect-related configuration